src/data/finsupp/lex.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.pi.lex
+import data.finsupp.order
+
+/-!
+# Lexicographic order on finitely supported functions
+
+This file defines the lexicographic order on `finsupp`.
+-/
+
+variables {α N : Type*}
+
+namespace finsupp
+
+lemma filter_ne_eq_empty_iff [decidable_eq α] [decidable_eq N] [has_zero N] {f g : α →₀ N} :
+  (f.support ∪ g.support).filter (λ a, f a ≠ g a) = ∅ ↔ f = g :=
+begin
+  refine ⟨λ h, _, λ h, h ▸ by simp⟩,
+  ext a,
+  contrapose h,
+  refine finset.ne_empty_of_mem (_ : a ∈ _),
+  simp only [ne.def, finset.mem_filter, finset.mem_union, finsupp.mem_support_iff],
+  exact ⟨not_and_distrib.1 $ mt (and_imp.2 eq.substr) (ne.symm h), h⟩,
+end
+
+section N_has_zero
+variables [has_zero N]
+
+/-- The lexicographic relation on `α →₀ N`, where `α` is ordered by `r`,
+  and `N` is ordered by `s`. -/
+protected def lex (r : α → α → Prop) (s : N → N → Prop) (x y : α →₀ N) : Prop :=
+∃ i, (∀ j, r j i → x j = y j) ∧ s (x i) (y i)
+
+lemma lex_eq_lex (r : α → α → Prop) (s : N → N → Prop) (x y : α →₀ N) :
+  finsupp.lex r s x y = pi.lex r (λ _, s) x y := rfl
```

Comment:
I thought you said these weren't defeq!

Reply:
Oh, I think that I tried `pi.lex r s x y`, got an error and did not think of fixing it!

Reply:
Anyway, the definition is the `pi.lex` application now... :upside_down_face: 

Sorry for misleading you!